### DISCUSSION TITLE
Does such a tree of subfolders on the panel take up a lot of RAM?

### DISCUSSION THREAD

**Discussion by korenevskiy** [Chris2011/NetBeans-Breadcrumb-Explorer#10] [Ideas]
Thanks, I installed it. It looks beautiful. And very convenient. Thank you.
When I test, a list of subfolders and files is displayed there. When I hover the mouse over a subfolder, it also displays a list of subfolders, and when I hover over one of the subfolders, a list of subfolders is also displayed. If I have 2000 files in my project and 30-40 files are open in tabs. Does such a tree of subfolders on the panel take up a lot of RAM?

**Comment by Chris2011** [comment]
If you have 2000+ files inside your project, nothing happens, when you open NetBeans. All the folders and subfolders, and subfolders and so on are lazy loaded, measn they just get the files and folders on clicking. I hope that Java garbage collection is smart enough to run when the JDialogs are not visible anymore, but I dunno yet. You can just test this for me which would be great and if there is any ram problem I will try to fix them by adding logig to reduce amount of data on destroying the JDialogs.

**Reply by Chris2011** [reply]
If you just open the files and you never use the breadcrumb, nothing should happen with your RAM.

**Comment by korenevskiy** [comment]
I don't know how to measure, but I just looked at the volume of one project.
11 thousand files. I just didn't know about it, I have 30 such projects.
The IDE is stable, as it was before. Of course, I have not tested the opening of directory trees for each directory, in the amount of 1000 files each. But a person is not able to view files in such a volume.

Thank you. The discussion can be closed.

**Reply by Chris2011** [reply]
If it is a HTML/JS/Node project with npm, that could be the thing that you have a lot of files. But as I said nothing is loaded for each file, it is lazy loaded. First you open the file in the editor, it will just render the breadcrumb, nothing else. If you click on a folder it will load all subfolders and files. Bust just the level that you opened, no children, grandchildren, grandgrandchildren, etc... So yes it could be, that you open 40-50 files and if you don't encounter a memory problem, thats fine. Opening 11000 files I guess this will not work in general of the IDE.
